Alluvial/Lease.cs

```diff
@@ -12,17 +12,22 @@ public class Lease : ILease
     {
         private readonly CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
         private readonly Func<TimeSpan, Task<TimeSpan>> extend;
+        private readonly Func<Task> release;
         private TimeSpan duration;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="Lease"/> class.
         /// </summary>
         /// <param name="duration">The duration of the lease.</param>
         /// <param name="extend">A delegate that will be called if the lease is extended.</param>
-        public Lease(TimeSpan duration, Func<TimeSpan, Task<TimeSpan>> extend = null)
+        public Lease(
+            TimeSpan duration, 
+            Func<TimeSpan, Task<TimeSpan>> extend = null,
+            Func<Task> release = null)
```

Comment:
It's intentional that `Backoff` hasn't changed. It may still change, but for the moment it didn't seem necessary.

The reason to not release the lease is to give control over when it does get released to the developer and default to the presumably safer full expiration time as we discussed before. Presumably they'll release it sooner, but they could also, for example, implement a cooldown / wait interval before the lease becomes available again. This achieve your original suggestion of reducing the lease term once work is done, e.g.:

```csharp

distributor.OnReceive(async (lease, next) => 
{
    await (next);
    Task.Delay(TimeSpan.FromSeconds(30)).ContinueWith(_ => lease.Release());
});

```